### ISSUE TITLE
Courtesy Accidentals

### ISSUE THREAD

**Issue by AlbertHart** [opensheetmusicdisplay/opensheetmusicdisplay#747]
In working with the OSMD Demo, and An die ferne Geliebte, I noticed that the musicxml file has courtesy accidentals in measure 12, which do not display in the OSMD Demo.

Is that an option somewhere, or a bug?

Here is the courtesy accidental in the MusicXML file

```
 <measure number="12" width="209.53">
      <note default-x="16.27" default-y="-181.96">
        <pitch>
          <step>E</step>
          <alter>-1</alter>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>eighth</type>

        <accidental>flat</accidental>


        <stem>up</stem>
        <staff>1</staff>
        </note>
```

And here is the rendering in MuseScore and OSMD Demo

![image](https://user-images.githubusercontent.com/8965352/82346641-f2355a80-99b3-11ea-9be9-3b740142dabc.png)

**Comment by sschmidTU** [comment]
Good catch! We don't do this yet because of the way our accidental system works, but it would be nice to have, ideally with an option whether to show courtesy accidentals, default yes.

**Comment by AlbertHart** [comment]
You must go out of your way to identify a courtesy accidental - since MusicXML just treats them like any other accidental. Since they are already in the MusicXML file, I wouldn't bother making an option for them. (But it would be if you do decide to make it an option.)

As a side note, I see that Sibelius just adds them automatically. In the image below, I entered the same notes into measures 1 and 5, then when I changed the Eb to a E-natural in measure 5, Sibelius automatically added the courtesy Flat symbol to measure 6. MuseScore does not add them automatically.

![image](https://user-images.githubusercontent.com/8965352/82360563-d12a3500-99c6-11ea-8476-70943a8ee60a.png)

**Comment by sschmidTU** [comment]
commit 8357fd4e9c606bd7f9a3e364798cd927df5935c9 enabled showing some courtesy accidentals given.

Unfortunately, in rare cases, we add courtesy accidentals which are unnecessary, like a natural before an E that never had a sharp or flat before.
like in Mozart Clarinet Quintet, measure 14:
![image](https://user-images.githubusercontent.com/33069673/83156985-5fa76080-a103-11ea-8a6f-6699abdf0906.png)

also in Dichterliebe measure 12:
![image](https://user-images.githubusercontent.com/33069673/83156891-42729200-a103-11ea-9c35-1ca66043f650.png)

but in most cases, the change was good in newly showing correct courtesy accidentals,
like in An die Ferne Geliebte, measure 8:
![image](https://user-images.githubusercontent.com/33069673/83157051-7483f400-a103-11ea-8147-b11d8a53a5a2.png)


So maybe, we can just give an option to not show courtesy accidentals.

**Comment by AlbertHart** [comment]
Has this been updated into the OSMD Demo?
Do things get updated automatically into the Demo?
It does seem to be working well in the OSMD Demo.

This is measures 6 to 8 of An Die Ferne Geleibte in the demo, and the courtesy accidental seems to have been added automatically. (It is not in the MusicXML file - in the second image.)

![image](https://user-images.githubusercontent.com/8965352/83770011-c960c680-a63d-11ea-838b-638e3acacef3.png)

![image](https://user-images.githubusercontent.com/8965352/83770057-d8e00f80-a63d-11ea-83b9-b02d90c4f042.png)

**Comment by sschmidTU** [comment]
The local demo (npm start) follows the updates of the branch, usually develop.
The [hosted demo](https://opensheetmusicdisplay.github.io/demo/) gets updated when there's a new master branch tag (usually at a new OSMD release).

**Comment by sschmidTU** [comment]
I just removed many of the unnecessary extra courtesy accidentals that were added in 8357fd4e9c606bd7f9a3e364798cd927df5935c9.

like in Mozart Clarinet Quintet, measure 9 and 13:
![image](https://user-images.githubusercontent.com/33069673/83156985-5fa76080-a103-11ea-8a6f-6699abdf0906.png)

**Comment by sschmidTU** [comment]
Fixed.

<img width="172" alt="Image" src="https://github.com/user-attachments/assets/932527e1-5b62-4b09-b48e-15240615261c" />

(An die Ferne measure 12)
